drivers/sensor/pmw3360/pmw3360.c

```diff
@@ -0,0 +1,938 @@
+
+	struct sensor_trigger trig = {
+		.type = SENSOR_TRIG_DATA_READY,
+		.chan = SENSOR_CHAN_ALL,
+	};
+	handler(DEVICE_GET(pmw3360), &trig);
+}
+
+static int pmw3360_async_init_power_up(struct pmw3360_data *dev_data)
+{
+	/* Reset sensor */
+
+	spi_cs_ctrl(dev_data, false);
+
+	return reg_write(dev_data, PMW3360_REG_POWER_UP_RESET, 0x5A);
+}
+
+static int pmw3360_async_init_configure(struct pmw3360_data *dev_data)
+{
+	int err;
+
+	err = update_cpi(dev_data, CONFIG_PMW3360_CPI);
+
+	if (!err) {
+		err = update_downshift_time(dev_data,
+					    PMW3360_REG_RUN_DOWNSHIFT,
+					    CONFIG_PMW3360_RUN_DOWNSHIFT_TIME_MS);
+	}
+
+	if (!err) {
+		err = update_downshift_time(dev_data,
+					    PMW3360_REG_REST1_DOWNSHIFT,
+					    CONFIG_PMW3360_REST1_DOWNSHIFT_TIME_MS);
+	}
+
+	if (!err) {
+		err = update_downshift_time(dev_data,
+					    PMW3360_REG_REST2_DOWNSHIFT,
+					    CONFIG_PMW3360_REST2_DOWNSHIFT_TIME_MS);
+	}
+
+	if (!err) {
+		dev_data->ready = true;
+	}
+
+	return err;
+}
+
+static void pmw3360_async_init(struct k_work *work)
+{
+	struct pmw3360_data *dev_data = &pmw3360_data;
+
+	ARG_UNUSED(work);
+
+	LOG_DBG("PMW3360 async init step %d", dev_data->async_init_step);
+
+	dev_data->err = async_init_fn[dev_data->async_init_step](dev_data);
+	if (dev_data->err) {
+		LOG_ERR("PMW3360 initailization failed");
+	} else {
+		dev_data->async_init_step++;
+
+		if (dev_data->async_init_step == ASYNC_INIT_STEP_COUNT) {
+			dev_data->ready = true;
+			LOG_INF("PMW3360 initialized");
+		} else {
+			k_delayed_work_submit(&dev_data->init_work,
+					      async_init_delay[dev_data->async_init_step]);
+		}
+	}
+}
+
+static int pmw3360_init_cs(struct pmw3360_data *dev_data)
+{
+	int err;
+
+	dev_data->cs_gpio_dev =
+		device_get_binding(CONFIG_PMW3360_CS_GPIO_DEV_NAME);
+	if (!dev_data->cs_gpio_dev) {
+		LOG_ERR("Cannot get CS GPIO device");
+		return -ENXIO;
+	}
+
+	err = gpio_pin_configure(dev_data->cs_gpio_dev,
+				 CONFIG_PMW3360_CS_GPIO_PIN,
+				 GPIO_DIR_OUT);
+	if (err) {
+		LOG_ERR("Cannot configure CS PIN");
+	}
+
+	return err;
+}
+
+static int pmw3360_init_irq(struct pmw3360_data *dev_data)
+{
+	int err;
+
+	dev_data->irq_gpio_dev =
+		device_get_binding(CONFIG_PMW3360_IRQ_GPIO_DEV_NAME);
+	if (!dev_data->irq_gpio_dev) {
+		LOG_ERR("Cannot get IRQ GPIO device");
+		return -ENXIO;
+	}
+
+	err = gpio_pin_configure(dev_data->irq_gpio_dev,
+				 CONFIG_PMW3360_IRQ_GPIO_PIN,
+				 GPIO_DIR_IN | GPIO_INT | GPIO_PUD_PULL_UP |
+				 GPIO_INT_LEVEL | GPIO_INT_ACTIVE_LOW);
+	if (err) {
+		LOG_ERR("Cannot configure IRQ GPIO");
+		return err;
+	}
+
+	gpio_init_callback(&dev_data->irq_gpio_cb, irq_handler,
+			   BIT(CONFIG_PMW3360_IRQ_GPIO_PIN));
+
+	err = gpio_add_callback(dev_data->irq_gpio_dev, &dev_data->irq_gpio_cb);
+	if (err) {
+		LOG_ERR("Cannot add IRQ GPIO callback");
+	}
+
+	return err;
+}
+
+static int pmw3360_init_spi(struct pmw3360_data *dev_data)
+{
+	dev_data->spi_dev = device_get_binding(DT_SPI_1_NAME);
```

Comment:
Fixups I guess. This part of code was created 9 months ago and there were many changes to DT since there. Anyway this still works:
```
$ grep -r DT_SPI_1_NAME build-nrf52840_pca20041/
build-nrf52840_pca20041/zephyr/include/generated/generated_dts_board_fixups.h:#define DT_SPI_1_NAME                     DT_NORDIC_NRF_SPI_SPI_1_LABEL
```

However this is wrong as this name must be defined in Kconfig so enduser could connect sensor to other SPI.

Reply:
SPI device name added to Kconfig.